contracts/multisig-prover/src/encoding/abi.rs

```diff
@@ -1,189 +1,136 @@
+    let msg = Keccak256::digest(encode(&command_batch.data).as_slice());
 
-        let id = BatchID::new(&self.message_ids, self.new_worker_set);
+    // Prefix for standard EVM signed data https://eips.ethereum.org/EIPS/eip-191
+    let unsigned = [
+        "\x19Ethereum Signed Message:\n32".as_bytes(), // Keccek256 hash length = 32
+        msg.as_slice(),
+    ]
+    .concat();
 
-        Ok(CommandBatch {
-            id,
-            message_ids: self.message_ids,
-            data,
-        })
-    }
+    Keccak256::digest(unsigned).as_slice().into()
 }
 
-impl CommandBatch {
-    pub fn msg_to_sign(&self) -> HexBinary {
-        let msg = Keccak256::digest(self.data.encode().as_slice());
+pub fn encode_execute_data(
+    command_batch: &CommandBatch,
+    quorum: Uint256,
+    signers: Vec<(Signer, Option<Signature>)>,
+) -> Result<HexBinary, ContractError> {
+    let param = ethabi::encode(&[
+        Token::Bytes(encode(&command_batch.data).into()),
+        Token::Bytes(encode_proof(quorum, signers)?.into()),
+    ]);
 
-        // Prefix for standard EVM signed data https://eips.ethereum.org/EIPS/eip-191
-        let unsigned = [
-            "\x19Ethereum Signed Message:\n32".as_bytes(), // Keccek256 hash length = 32
-            msg.as_slice(),
-        ]
-        .concat();
+    let input = ethabi::encode(&[Token::Bytes(param)]);
 
-        Keccak256::digest(unsigned).as_slice().into()
-    }
+    let mut calldata = short_signature(GATEWAY_EXECUTE_FUNCTION_NAME, &[ParamType::Bytes]).to_vec();
 
-    pub fn encode_execute_data(
-        &self,
-        quorum: Uint256,
-        signers: Vec<(Signer, Option<Signature>)>,
-    ) -> Result<HexBinary, ContractError> {
-        let param = ethabi::encode(&[
-            Token::Bytes(self.data.encode().into()),
-            Token::Bytes(self.encode_proof(quorum, signers)?.into()),
-        ]);
+    calldata.extend(input);
 
-        let input = ethabi::encode(&[Token::Bytes(param)]);
+    Ok(calldata.into())
+}
 
-        let mut calldata =
-            short_signature(GATEWAY_EXECUTE_FUNCTION_NAME, &[ParamType::Bytes]).to_vec();
+fn encode_proof(
+    quorum: Uint256,
+    signers: Vec<(Signer, Option<Signature>)>,
+) -> Result<HexBinary, ContractError> {
+    let mut operators = make_operators(signers)?;
+    operators.sort();
 
-        calldata.extend(input);
+    let (addresses, weights, signatures): (Vec<_>, Vec<_>, Vec<_>) = operators
+        .into_iter()
+        .map(|op| {
+            (
+                Token::Address(ethereum_types::Address::from_slice(op.address.as_slice())),
+                Token::Uint(ethereum_types::U256::from_big_endian(
+                    &op.weight.to_be_bytes(),
+                )),
+                op.signature.map(|sig| Token::Bytes(<Vec<u8>>::from(sig))),
+            )
+        })
+        .multiunzip();
 
-        Ok(calldata.into())
-    }
+    let signatures = signatures.into_iter().flatten().collect();
```

Comment:
nice! Didn't know flatten would do that for options 👍

Reply:
clippy actually caught this